src/dwarf/resolver.rs

```diff
@@ -22,14 +25,79 @@ use super::parser::DWSymInfo;
 use super::parser::DebugLineCU;
 
 
+impl From<&DWSymInfo<'_>> for SymInfo {
+    fn from(other: &DWSymInfo) -> Self {
+        let DWSymInfo {
+            name,
+            addr,
+            size,
+            sym_type,
+            ..
+        } = other;
+
+        SymInfo {
+            name: name.to_string(),
+            addr: *addr as Addr,
+            size: *size,
+            sym_type: *sym_type,
+            file_offset: 0,
+            obj_file_name: None,
+        }
+    }
+}
+
+
+struct Cache<'mmap> {
+    /// The ELF parser object used for reading data.
+    parser: &'mmap ElfParser,
+    /// Debug symbols, ordered by symbol name.
+    debug_syms: Option<Vec<DWSymInfo<'mmap>>>,
+}
+
+impl<'mmap> Cache<'mmap> {
+    /// Create a new `Cache` using the provided ELF parser.
+    fn new(parser: &'mmap ElfParser) -> Self {
+        Self {
+            parser,
+            debug_syms: None,
+        }
+    }
+
+    /// Extract the symbol information from DWARf if having not done it before.
```

Comment:
is the mix of `///` and `//` intentional? also mixed casing in "DWARf"

Reply:
`///` comments are doc comments and appear in the rendered documentation. I didn't want to include the `Note` part in there, as it's really mostly an implementation comment, hence the two kinds.